src/core/tsi/ssl/ssl_transport_security.h

```diff
@@ -165,6 +168,28 @@ tsi_result tsi_ssl_client_handshaker_factory_create_handshaker(
     tsi_ssl_client_handshaker_factory* self, const char* server_name_indication,
     tsi_handshaker** handshaker);
 
+/* Creates a TLS handshaker.
+  - server_name_indication indicates the name of the server the client is
+    trying to connect to which will be relayed to the server using the SNI
+    extension.
+  - session_cache: SSL session cache for sessions resumption.
+  - options: TLS credential options instance.
+  - reload_arg: credential reload arg instance to be passed to API's in config.
+  - is_client: a boolean flag indicating if the handshaker will be used at
+    the client (is_client = true) or server (is_client = false) side.
+  - handshaker is the address of the handshaker pointer to be created.
+
+  - This method returns TSI_OK on success or TSI_INVALID_PARAMETER in the case
+    where a parameter is invalid. Note that some fields of handshaker will be
+    populated once credential reload is finished. */
+
+tsi_result tls_tsi_handshaker_create(const char* server_name_indication,
```

Comment:
should call tsi_tls_handshaker_create, where namespace is tsi. Same for rest of this file.